app/src/main/java/ch/heigvd/pro/b04/android/poll/PollAdapter.kt

```diff
@@ -0,0 +1,98 @@
+package ch.heigvd.pro.b04.android.poll
+
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import ch.heigvd.pro.b04.android.datamodel.Question
+import ch.heigvd.pro.b04.android.R
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import java.util.*
+
+class PollAdapter(private val state: PollViewModel,
+                  private val lifecycleOwner: LifecycleOwner
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    private var questions: List<Question> = LinkedList()
+    override fun getItemId(position: Int): Long {
+        return if (position == 0) HEADER_ID else questions[position - 1].idQuestion
+    }
+
+    private class HeaderViewHolder(parent: ViewGroup,
+                                   state: PollViewModel,
+                                   lifecycleOwner: LifecycleOwner
+    ) : RecyclerView.ViewHolder(LayoutInflater.from(parent.context)
+            .inflate(R.layout.poll_title, parent, false)) {
+
+        private val title: TextView = itemView.findViewById(R.id.poll_title)
+
+        init {
+            lifecycleOwner.lifecycleScope.launchWhenStarted {
+                state.poll.collect { poll -> title.text = poll.title }
+            }
```

Comment:
Since we have the `state` `PollViewModel`, we should possibly use the `viewModelScope` here.